dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -923,17 +964,60 @@ static bool parseFileName(const char *name,StringBuffer &mname,unsigned &num,uns
         name = nonrepdir.str();
     num = 0;
     max = 0;
+    unsigned dirPerPart = 0;
     for (;;) {
         char c=*name;
         if (!c)
             break;
+        if ((c=='d')&&(isdigit(name[1])))
```

Comment:
This routine should really parse the filename, looking for a plane prefix match first.
Reject if no match. All files should exist in a plane.

Then check next segment for striping, and check if corresponding plane has striping, if it hasn't then complain/reject.
Then parse remainder of path, and check penultimate segment for dir-per-part pattern (i.e. only check path segment before the filename).

Because this is called a lot, it does need to be efficient.
So I would collect the data plane prefixes upfront, then initially check to see if the name has a prefix of one of them.